File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/AuthorApi.java
package ru.elenacherev.librarymanager.api;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import ru.elenacherev.librarymanager.api.dto.Author;
import ru.elenacherev.librarymanager.api.dto.Edition;

import java.util.List;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

public interface AuthorApi {
    Author getAuthor(
    		@NotNull Long id
    );

    Author createAuthor(
    		@Valid Author author
    );

    Author saveAuthor(
    		@NotNull Long id, 
    		@Valid Author author
    );

    Page<Author> searchAuthors(
    		Pageable pageable
    );

    Page<Edition> searchEditions(
    		@NotNull Long id,  
    		Pageable pageable 
    );
    
    List<Edition> updateEditionsByAuthorId(
    		@NotNull Long authorid,
    		@NotNull List<Long> editionIds
    );
}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/AuthorApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Author
import ru.elenacherev.librarymanager.api.dto.Edition
import java.util.*
import javax.validation.Valid

interface AuthorApi {
    fun getAuthor(
        authorId: UUID
    ): Author?

    fun createAuthor(
        author: @Valid Author
    ): Author

    fun saveAuthor(
        authorId: UUID,
        author: @Valid Author
    ): Author

    fun searchAuthors(
        pageable: Pageable
    ): Page<Author>

    fun searchEditions(
        authorId: UUID,
        pageable: Pageable
    ): Page<Edition>

    fun updateEditionsByAuthorId(
        authorId: UUID,
        editionIds: List<UUID>
    ): List<Edition>
}